File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

N = int(input())

stack = []

for test_case in range(0, N):
    order = list(map(str, input().split()))

    if order[0] == 'push':
        stack.append(int(order[1]))
    elif order[0] == 'pop':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack.pop(-1))
    elif order[0] == 'size':
        print(len(stack))
    elif order[0] == 'empty':
        if len(stack) == 0:  # 스택이 비었는지 아닌지 유무를 확인하기 위해서는 stack == [] 대신 len(stack)을 사용하자.
            print(1)
        else:
            print(0)
    elif order[0] == 'top':
        if len(stack) == 0:
            print(-1)
        else:
            print(stack[-1])

File: 백준/Silver/10828. 스택/스택.cc
/*
stack

그냥 스택 구현하기
*/

#include <iostream>
#include <string>
#include <stack>

using namespace std;

int main() {
	int n;
	cin >> n;
	stack<int> arr;

	for (int i = 0; i < n; i++) {
		string order;
		int tmp;
		cin >> order;

		if (order == "push") {
			cin >> tmp;
			arr.push(tmp);
		}
		else if (order == "top") {
			if (arr.size() == 0) {
				cout << -1 << "\n";
			}
			else {
				cout << arr.top() << "\n";
			}
		}
		else if (order == "size") {
			cout << arr.size() << "\n";
		}
		else if (order == "empty") {
			if (arr.empty()) {
				cout << 1 << "\n";
			}
			else {
				cout << 0 << "\n";
			}
		}
		else if (order == "pop") {
			if (arr.size() == 0) {
				cout << -1 << "\n";
			}
			else {
				cout << arr.top() << "\n";
				arr.pop();
			}
		}
	}
	

	return 0;
}